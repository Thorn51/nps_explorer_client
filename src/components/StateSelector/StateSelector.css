/* Dropdown selector for states on homepage */
.state_selection {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 90%;
  color: var(--darkGunmetal);
  margin: 15px auto 15px auto;
}

.selected_state {
  text-align: center;
  font-size: 1.5em;
}

.state_selector_label {
  font-size: 0.9em;
  font-weight: bold;
  text-align: center;
  width: 100%;
}
