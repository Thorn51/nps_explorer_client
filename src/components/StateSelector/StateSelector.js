import React, { useContext } from "react";
import { GlobalContext } from "../../context/GlobalState";

export default function StateSelector() {
  // Get actions from context
  const { selectState, getParks } = useContext(GlobalContext);

  // On select state send and store state info to state, fetch parks
  const onChange = e => {
    e.preventDefault();
    let stateName = e.target.options[e.target.selectedIndex].text;
    let stateCode = e.target.value;

    // Store state information in state
    selectState(stateCode, stateName);

    // Fetch list of parks by state
    getParks(stateCode);
  };

  return (
    <div className="state_selection">
      <label htmlFor="states">What state would you like to explore?</label>
      <select name="states" id="state" onChange={onChange}>
        <option value="null">Select a State...</option>
        <option value="AL">Alabama</option>
        <option value="AK">Alaska</option>
        <option value="AZ">Arizona</option>
        <option value="AR">Arkansas</option>
        <option value="CA">California</option>
        <option value="CO">Colorado</option>
        <option value="CT">Connecticut</option>
        <option value="DE">Delaware</option>
        <option value="DC">District Of Columbia</option>
        <option value="FL">Florida</option>
        <option value="GA">Georgia</option>
        <option value="HI">Hawaii</option>
        <option value="ID">Idaho</option>
        <option value="IL">Illinois</option>
        <option value="IN">Indiana</option>
        <option value="IA">Iowa</option>
        <option value="KS">Kansas</option>
        <option value="KY">Kentucky</option>
        <option value="LA">Louisiana</option>
        <option value="ME">Maine</option>
        <option value="MD">Maryland</option>
        <option value="MA">Massachusetts</option>
        <option value="MI">Michigan</option>
        <option value="MN">Minnesota</option>
        <option value="MS">Mississippi</option>
        <option value="MO">Missouri</option>
        <option value="MT">Montana</option>
        <option value="NE">Nebraska</option>
        <option value="NV">Nevada</option>
        <option value="NH">New Hampshire</option>
        <option value="NJ">New Jersey</option>
        <option value="NM">New Mexico</option>
        <option value="NY">New York</option>
        <option value="NC">North Carolina</option>
        <option value="ND">North Dakota</option>
        <option value="OH">Ohio</option>
        <option value="OK">Oklahoma</option>
        <option value="OR">Oregon</option>
        <option value="PA">Pennsylvania</option>
        <option value="RI">Rhode Island</option>
        <option value="SC">South Carolina</option>
        <option value="SD">South Dakota</option>
        <option value="TN">Tennessee</option>
        <option value="TX">Texas</option>
        <option value="UT">Utah</option>
        <option value="VT">Vermont</option>
        <option value="VA">Virginia</option>
        <option value="WA">Washington</option>
        <option value="WV">West Virginia</option>
        <option value="WI">Wisconsin</option>
        <option value="WY">Wyoming</option>
      </select>
    </div>
  );
}
