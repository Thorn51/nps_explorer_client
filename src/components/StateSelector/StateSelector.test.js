import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import StateSelector from "./StateSelector";

describe("<StateSelector />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<StateSelector />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<StateSelector />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
