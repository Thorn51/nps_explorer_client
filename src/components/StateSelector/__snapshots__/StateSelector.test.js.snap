// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StateSelector /> Renders the UI as expected 1`] = `"tree"`;
